app/src/organisms/ProtocolSetupParameters/index.tsx

```diff
@@ -151,11 +163,56 @@ export function ProtocolSetupParameters({
     },
   })
   const handleConfirmValues = (): void => {
-    if (
-      enableCsvFile &&
-      mostRecentAnalysis?.result === 'parameter-value-required'
-    ) {
-      makeSnackbar(t('protocol_requires_csv') as string)
+    if (enableCsvFile) {
+      if (hasMissingFileParam) {
+        makeSnackbar(t('protocol_requires_csv') as string)
+      } else {
+        const dataFilesForProtocolMap = runTimeParametersOverrides.reduce<
+          Record<string, FileData>
+        >((acc, parameter) => {
+          // create {variableName: FileData} map for sending to /dataFiles endpoint
+          if (
+            parameter.type === 'csv_file' &&
+            parameter.file?.id == null &&
+            parameter.file?.file != null
+          ) {
+            return { [parameter.variableName]: parameter.file.file }
+          } else if (
+            parameter.type === 'csv_file' &&
+            parameter.file?.id == null &&
+            parameter.file?.filePath != null
+          ) {
+            return { [parameter.variableName]: parameter.file.filePath }
+          }
+          return acc
+        }, {})
+        Promise.all(
```

Comment:
probably we need to void

```suggestion
        void Promise.all(
```